cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -822,71 +822,55 @@ func (pm *PipelineManager) savePipelineProviderToEnv(
 	return nil
 }
 
+func validateProvider(provider ciProviderType) error {
+	if provider == "" {
+		errMsg := "Unknown CI provider"
+		log.Println("Error:", errMsg)
+		return fmt.Errorf(errMsg)
+	}
+	return nil
+}
+
 func (pm *PipelineManager) checkAndPromptForProviderFiles(
 	ctx context.Context, props projectProperties) error {
 	log.Printf("Checking for provider files for: %s", props.CiProvider)
 
-	providerFileChecks := map[ciProviderType]struct {
-		ymlPath             string
-		dirPath             string
-		dirDisplayName      string
-		providerDisplayName string
-	}{
-		ciProviderGitHubActions: {
-			ymlPath:             filepath.Join(props.RepoRoot, gitHubYml),
-			dirPath:             filepath.Join(props.RepoRoot, gitHubWorkflowsDirectory),
-			dirDisplayName:      gitHubWorkflowsDirectory,
-			providerDisplayName: gitHubDisplayName,
-		},
-		ciProviderAzureDevOps: {
-			ymlPath:             filepath.Join(props.RepoRoot, azdoYml),
-			dirPath:             filepath.Join(props.RepoRoot, azdoPipelinesDirectory),
-			dirDisplayName:      azdoPipelinesDirectory,
-			providerDisplayName: azdoDisplayName,
-		},
-	}
-
-	providerCheck, exists := providerFileChecks[props.CiProvider]
-	if !exists {
-		errMsg := fmt.Sprintf("%s is not a known pipeline provider", props.CiProvider)
-		log.Println("Error:", errMsg)
-		return fmt.Errorf(errMsg)
+	if err := validateProvider(props.CiProvider); err != nil {
+		return err
 	}
 
-	log.Printf("YAML path: %s", providerCheck.ymlPath)
-	log.Printf("Directory path: %s", providerCheck.dirPath)
+	provider := string(props.CiProvider)
```

Comment:
do we really need to cast here?  If this is for calling `log.Printf()`, you can pass props.CiProvider directly

Reply:
Removed